### PR TITLE
Don't include None in remote_json map

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-1784

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 1]
```diff
@@ -213,7 +214,10 @@ def _get_remote_urls_helper(self, koji_build):
             elif archive['type_name'] == 'json' and archive['filename'] == 'remote-source.json':
                 cachito_json_url = os.path.join(remote_sources_path, archive['filename'])
 
-        return remote_sources_urls, {dest_name: cachito_json_url}
+        if cachito_json_url and dest_name:
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py:217]: Can you explain me how dest_name works here? How is that aligned with json file?

Comment (MartinBasti) [reply atomic_reactor/plugins/pre_fetch_sources.py:217]: Unittest?

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 2]
```diff
@@ -213,7 +214,14 @@ def _get_remote_urls_helper(self, koji_build):
             elif archive['type_name'] == 'json' and archive['filename'] == 'remote-source.json':
                 cachito_json_url = os.path.join(remote_sources_path, archive['filename'])
 
-        return remote_sources_urls, {dest_name: cachito_json_url}
+        # we will add entry to remote_json_map only when
+        # build has remote sources archive and json
+        # it will be used later to get correspondent remoete-source.json for each remote sources
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py:219]: typo `remoete`

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 1 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_fetch_sources.py:221]: Can actually be `cachito_json_url XOR dest_name == True` a valid case? Should we raise exception if only one is set?

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 3]
```diff
@@ -213,7 +214,23 @@ def _get_remote_urls_helper(self, koji_build):
...
+        if cachito_json_url and dest_name:
+            remote_json_map = {dest_name: cachito_json_url}
+
+            if len(remote_sources_urls) > 1:
+                raise RuntimeError('There can be just one remote sources archive, got : {}'.
+                                   format(remote_sources_urls))
+
+        elif bool(cachito_json_url) != bool(dest_name):
+            raise RuntimeError('Remote sources archive or remote source json missing, '
+                               'remote source archiv: {}, remote source json: {}'.
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_fetch_sources.py:230]: typo `archiv`

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 2 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: fixed

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 3 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: fixed

File: atomic_reactor/plugins/pre_fetch_sources.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py:221]: added unitests

**Comment by rcerven** [comment]
rn not updated, fix for unreleased feature

**Pull request merged**